From 1ad3c286ff1c9c170fb1c2cb576e1ede2fd6a25e Mon Sep 17 00:00:00 2001
From: "tao.zeng" <tao.zeng@amlogic.com>
Date: Tue, 22 Jul 2014 17:33:04 +0800
Subject: [PATCH 4558/5965] PD #83628: driver update for AML1218, version
 0.92_C

1. fix bug of set long press threshold;
2. update code version to 0.92_C;

Change-Id: I66750bd19ec732fccd87dff6b37c9b7354546e19
---
 drivers/amlogic/power/aml_pmu/aml1218-sply.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/power/aml_pmu/aml1218-sply.c b/drivers/amlogic/power/aml_pmu/aml1218-sply.c
index ffc6062d8575..0d5402179259 100755
--- a/drivers/amlogic/power/aml_pmu/aml1218-sply.c
+++ b/drivers/amlogic/power/aml_pmu/aml1218-sply.c
@@ -510,6 +510,7 @@ int aml1218_set_long_press_time(int ms)
      
     aml1218_read16(0x90, &val);
     tmp = ms/100 -1; 
+    val &= ~0x7f;
     val |= tmp;                                        // set power key long press to 10s
     return aml1218_set_bits(0x0090, val, 0x7f);
 }
@@ -1508,7 +1509,7 @@ static int aml1218_battery_probe(struct platform_device *pdev)
     int      ret;
     uint32_t tmp2;
 
-	AML1218_INFO("---> PMU driver version:v0.91_B\n");
+	AML1218_INFO("---> PMU driver version:v0.92_C\n");
 	AML1218_DBG("call %s in", __func__);
     g_aml1218_init = pdev->dev.platform_data;
     if (g_aml1218_init == NULL) {
-- 
2.19.0

